src/auth/ldap_krb/common_utils.hpp

```diff
@@ -0,0 +1,436 @@
+  };
+
+  const auto is_open_or_close = [](const char c) {
+      return ((c == OPEN_OID) or (c == CLOSE_OID));
+  };
+
+
+  std::string str_trim(const std::string&);
+  auto from_const_str_to_char(const std::string&);
+  auto from_const_ptr_char_to_string(const char*);
+  auto duplicate_ptr_char(const char*str_to_convert);
+  uint32_t auth_trap_error(const uint32_t);
+  timeval from_chrono_to_timeval(const system_clock::time_point&);
+  system_clock::time_point from_timeval_to_chrono(const timeval& time_val);
+
+
+  template<typename Etp_>
+  using enable_enum_t =
+    std::enable_if_t<std::is_enum<Etp_>::value,
+                            std::underlying_type_t<Etp_>>;
+
+  template<typename Etp_>
+  constexpr inline enable_enum_t<Etp_>
+  enum_underlying_value(Etp_ enm) noexcept
+  {
+    return static_cast<std::underlying_type_t<Etp_>>(enm);
+  }
+
+  template<typename Enm, typename Etp_>
+  constexpr inline typename
+  std::enable_if_t<std::is_enum<Enm>::value &&
+                   std::is_integral<Etp_>::value, Enm>::type
+  convert_to_enum(Etp_ value) noexcept
+  {
+    return static_cast<Enm>(value);
+  }
+
+}   //-- namespace common_utils
+
+
+namespace timer {
+    using chrono_hi_resolution =
+    std::chrono::high_resolution_clock;
+    using chrono_tp_t          =
+    std::chrono::high_resolution_clock::time_point;
+
+    class Timer {
+    public:
+        Timer() { timer_start(); };
+
+        ~Timer() = default;
+
+        void timer_start() {
+          m_start = chrono_hi_resolution::now();
+        }
+
+        void timer_end() {
+          m_finish = chrono_hi_resolution::now();
+        }
+
+        std::chrono::nanoseconds elapsed_in_nano_secs() const {
+          using std::chrono::duration_cast;
+          using std::chrono::nanoseconds;
+          return duration_cast<nanoseconds>
+              (chrono_hi_resolution::now() - m_start);
+        }
+
+        std::chrono::microseconds elapsed_in_micro_secs() const {
+          using std::chrono::duration_cast;
+          using std::chrono::microseconds;
+          return duration_cast<microseconds>
+              (chrono_hi_resolution::now() - m_start);
+        }
+
+        std::chrono::milliseconds elapsed_in_milli_secs() const {
+          using std::chrono::duration_cast;
+          using std::chrono::milliseconds;
+          return duration_cast<milliseconds>
+              (chrono_hi_resolution::now() - m_start);
+        }
+        //-- TODO: Templatize elapsed_time
+
+
+    private:
+        chrono_tp_t m_start;
+        chrono_tp_t m_finish;
+
+        friend std::ostream& operator<<(std::ostream& out_stream,
+                                        const Timer& timer) {
+          const std::string MILLISEC_STR("ms");
+          return out_stream
+              << timer.elapsed_in_milli_secs().count()
+              << MILLISEC_STR;
+        }
+    };
+}
+
+/*  Just a simple attempt to get a thread pool implemented.
+*/
+namespace threading {
```

Comment:
this namespace, or its contents, don't seem to be used anywhere in this PR. Please remove.